cmd/setec/setec.go

```diff
@@ -35,97 +38,90 @@ import (
 )
 
 func main() {
-	serverCmd := &ffcli.Command{
-		Name:       "server",
-		ShortUsage: "setec server [flags]",
-		ShortHelp:  "run the setec server",
-		FlagSet: func() *flag.FlagSet {
-			fs := flag.NewFlagSet("server", flag.ExitOnError)
-			fs.StringVar(&serverArgs.StateDir, "state-dir", "", "setec state dir")
-			fs.StringVar(&serverArgs.Hostname, "hostname", "", "Tailscale hostname to use")
-			fs.StringVar(&serverArgs.KMSKeyName, "kms-key-name", "", "name of KMS key to use for database encryption")
-			fs.BoolVar(&serverArgs.Dev, "dev", false, "dev mode")
-			return fs
-		}(),
-		Exec: runServer,
-	}
-	listCmd := &ffcli.Command{
-		Name:       "list",
-		ShortUsage: "setec list [flags] <server>",
-		ShortHelp:  "list secrets",
-		Exec:       runList,
-	}
-	infoCmd := &ffcli.Command{
-		Name:       "info",
-		ShortUsage: "setec info [flags] <server> <secret-name>",
-		ShortHelp:  "get secret info",
-		Exec:       runInfo,
-	}
-	getCmd := &ffcli.Command{
-		Name:       "get",
-		ShortUsage: "setec get [flags] <server> <secret-name>",
-		ShortHelp:  "get a secret",
-		FlagSet: func() *flag.FlagSet {
-			fs := flag.NewFlagSet("get", flag.ExitOnError)
-			fs.BoolVar(&getArgs.IfChanged, "if-changed", false, "get active version if changed from --version")
-			fs.Uint64Var(&getArgs.Version, "version", 0, "secret version to retrieve (default: the active version)")
-			return fs
-		}(),
-		Exec: runGet,
-	}
-	putCmd := &ffcli.Command{
-		Name:       "put",
-		ShortUsage: "setec put [flags] <server> <secret-name>",
-		ShortHelp:  "put a secret",
-		FlagSet: func() *flag.FlagSet {
-			fs := flag.NewFlagSet("put", flag.ExitOnError)
-			fs.StringVar(&putArgs.File, "from-file", "", "read secret value from file instead of prompting interactively")
-			return fs
-		}(),
-		Exec: runPut,
-	}
-	setActiveCmd := &ffcli.Command{
-		Name:       "activate",
-		ShortUsage: "setec activate [flags] <server> <secret-name> <secret-version>",
-		ShortHelp:  "activate a secret version",
-		Exec:       runSetActive,
-	}
-
-	root := &ffcli.Command{
-		Name:       "setec",
-		ShortUsage: "setec <subcmd>",
-		Subcommands: []*ffcli.Command{
-			serverCmd,
-			listCmd,
-			infoCmd,
-			getCmd,
-			putCmd,
-			setActiveCmd,
+	root := &command.C{
+		Name:  filepath.Base(os.Args[0]),
+		Usage: "server [options]\ncommand [flags] ...\nhelp [command]",
+		Help: `A server and command-line tool for the setec API.
+
+The "server" subcommand starts a server for the setec API.
+The other subcommands call methods of a running setec server.`,
+
+		Commands: []*command.C{
+			{
+				Name:  "server",
+				Usage: "[flags]",
```

Comment:
This feels like it could be a default when Usage is unspecified?

Even fancier: unspecified usage could autogenerate from introspection of parameters, once positional args are bolted into it somehow, so if you define some positional args plus flags, Usage defaults to `[flags] pos-arg-1 pos-arg-2 variadic...` or somesuch?

Reply:
We can probably leave it out in this case, since the help will show the flags if they're defined anyway. 

I've thought about parsing the usage strings a little more to get positional args. Maybe that can be a way of doing it, e.g., we might recognize

```
<a> <b>       -- two positional args
<a> [<b> ...] -- one or more positional args, with a rest slice
<a> [<b>] <c> -- two (into v[0] and v[2]) or three (into v[0], v[1], v[2])
```

and provide a kind of scanf-style "unmarshaler".

Reply:
(I decided on a different approach, noted elsethread)